### PR TITLE
create tests for USB configuration in firmware

### PR DESCRIPTION
**Pull request by PrzemyslawBanasiak13**
Signed-off-by: pbanasiak <przemyslaw.banasiak@3mdeb.com>

### CODE REVIEW

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md [hunk 1]
```diff
@@ -405,3 +405,142 @@ Algorithm       Hash
...
+
+**Test description**
+
+This test aims to verify if USB stack can be enabled. USB stack controls loading
+of UEFI USB drivers.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `SETUP_MENU_KEY` = F2
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:419]: Are we sure, that this setup key will be correct for all Dasharo platforms?

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md [hunk 2]
```diff
@@ -405,3 +405,142 @@ Algorithm       Hash
...
+
+**Test description**
+
+This test aims to verify if USB stack can be disabled. USB stack controls
+loading of UEFI USB drivers.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `SETUP_MENU_KEY` = F2
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:453]: Are we sure, that this setup key will be correct for all Dasharo platforms?

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md [hunk 3]
```diff
@@ -405,3 +405,142 @@ Algorithm       Hash
 ---------       ----                                                              ----
 SHA256          F46597C0C63A1EEFB200D40EDF654E52F10C3D5D21565886AD603FABAF8D39FB  E:\test_file.txt
 ```
+
+## USB004.001 Enable USB stack (Firmware)
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:409]: I think these test cases should be localized at the top of the test suite (this is the basic functionality).

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md [hunk 4]
```diff
@@ -405,3 +405,142 @@ Algorithm       Hash
...
+1. Hover on checkbox next to `Enable USB stack` option, and make sure that `X`
+    sign is present between square brackets - `[X]`. Use `Spacebar to toggle.
+1. Save using `F10`, and exit using `Esc`.
+1. Reboot the device.
+1. While booting, press the `SETUP_MENU_KEY` to enter Setup Menu.
+1. Enter the `Dasharo System Features` menu option.
+
+**Expected result**
+
+1. `X` sign is present between square brackets next to `Enable USB stack` -
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:440]: The correctness of the function should not be determined by whether the item continues to be active after a reboot, but by whether it actually does not work (i. e. USBs are visualized in the Boot menu).

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md [hunk 5]
```diff
@@ -405,3 +405,142 @@ Algorithm       Hash
...
+    sign is NOT present between square brackets - `[ ]`. Use `Spacebar to
+    toggle.
+1. Save using `F10`, and exit using `Esc`.
+1. Reboot the device.
+1. While booting, press the `SETUP_MENU_KEY` to enter Setup Menu.
+1. Enter the `Dasharo System Features` menu option.
+
+**Expected result**
+
+1. `X` sign is NOT present between square brackets next to `Enable USB stack` -
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:475]: The correctness of the function should not be determined by whether the item continues to be active after a reboot, but by whether it actually does not work (i. e. USBs are not visualized in the Boot menu, the USB keyword on the firmware level does not work).

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md [hunk 6]
```diff
@@ -405,3 +405,142 @@ Algorithm       Hash
...
+    that `X` sign is present between square brackets - `[X]`. Use `Spacebar to
+    toggle.
+1. Save using `F10`, and exit using `Esc`.
+1. Reboot the device.
+1. While booting, press the `SETUP_MENU_KEY` to enter Setup Menu.
+1. Enter the `Dasharo System Features` menu option.
+
+**Expected result**
+
+1. `X` sign is present between square brackets next to
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:510]: The correctness of the function should not be determined by whether the item continues to be active after a reboot, but by whether it actually does not work (i. e. USBs are visualized in the Boot menu).

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md [hunk 7]
```diff
@@ -405,3 +405,142 @@ Algorithm       Hash
...
+    that `X` sign is NOT present between square brackets - `[ ]`. Use `Spacebar
+    to toggle.
+1. Save using `F10`, and exit using `Esc`.
+1. Reboot the device.
+1. While booting, press the `SETUP_MENU_KEY` to enter Setup Menu.
+1. Enter the `Dasharo System Features` menu option.
+
+**Expected result**
+
+1. `X` sign is NOT present between square brackets next to
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:545]: The correctness of the function should not be determined by whether the item continues to be active after a reboot, but by whether it actually does not work (i. e. USBs are not visualized in the Boot menu).

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md [hunk 8]
```diff
@@ -405,3 +405,142 @@ Algorithm       Hash
 ---------       ----                                                              ----
 SHA256          F46597C0C63A1EEFB200D40EDF654E52F10C3D5D21565886AD603FABAF8D39FB  E:\test_file.txt
 ```
+
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:408]: @PrzemyslawBanasiak13 should we improve the test matrix for any of Dasharo's products? As I know, this feature is dedicated only to MSI Z690-A

**Review by sulewskiprzemyslaw — CHANGES_REQUESTED** [review]

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md (hunk 1 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:419]: Any setup isn't valid for all the platforms. Should I remove that part?

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md (hunk 3 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:409]: Moved

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md (hunk 4 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:440]: fixed

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md (hunk 5 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:475]: Fixed

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md (hunk 7 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:545]: Fixed

File: docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md (hunk 8 above)
Comment (PrzemyslawBanasiak13) [reply docs/unified-test-documentation/dasharo-compatibility/306-usb-hid-and-msc-support.md:553]: Updated test matrices for Novacustom platforms

**Pull request merged**